src/main/java/org/broadinstitute/hellbender/tools/walkers/filters/VariantFiltration.java

```diff
@@ -0,0 +1,389 @@
+     */
+    private String possiblyInvertFilterExpression( final String description ){
+        return invertFilterExpression ? "Inverse of: " + description : description;
+    }
+
+    private void initializeVcfWriter() {
+
+        //XXX hardwired to output VCFs
+        //XXX unset indexing on the fly
+        writer = new VariantContextWriterBuilder().setOutputFile(out).setOutputFileType(VariantContextWriterBuilder.OutputType.VCF).unsetOption(Options.INDEX_ON_THE_FLY).build();
+
+        // setup the header fields
+        final Set<VCFHeaderLine> hInfo = new HashSet<>();
+        hInfo.addAll(getHeaderForVariants().getMetaDataInSortedOrder());
+
+        if ( clusterWindow > 0 ) {
+            hInfo.add(new VCFFilterHeaderLine(CLUSTERED_SNP_FILTER_NAME, "SNPs found in clusters"));
+        }
+
+        if ( !genotypeFilterExps.isEmpty() ) {
+            hInfo.add(VCFStandardHeaderLines.getFormatLine(VCFConstants.GENOTYPE_FILTER_KEY));
+        }
+
+        try {
+            for ( final JexlVCMatchExp exp : filterExps ) {
+                hInfo.add(new VCFFilterHeaderLine(exp.name, possiblyInvertFilterExpression(exp.exp.toString())));
+            }
+            for ( final JexlVCMatchExp exp : genotypeFilterExps ) {
+                hInfo.add(new VCFFilterHeaderLine(exp.name, possiblyInvertFilterExpression(exp.exp.toString())));
+            }
+
+            if ( mask != null ) {
+                if (filterRecordsNotInMask) {
+                    hInfo.add(new VCFFilterHeaderLine(maskName, "Doesn't overlap a user-input mask"));
+                } else {
+                    hInfo.add(new VCFFilterHeaderLine(maskName, "Overlaps a user-input mask"));
+                }
+            }
+        } catch (final IllegalArgumentException e) {
+            throw new UserException.BadInput(e.getMessage());
+        }
+
+        writer.writeHeader(new VCFHeader(hInfo, getHeaderForVariants().getGenotypeSamples()));
+    }
+
+    @Override
+    public void onTraversalStart() {
+        if ( maskExtension < 0 ) {
+            throw new UserException.BadArgumentValue("maskExtension", "negative values are not allowed");
+        }
+
+        if (filterRecordsNotInMask && mask == null) {
+            throw new UserException.BadArgumentValue("filterNotInMask", "argument not allowed if mask argument is not provided");
+        }
+        filterExps = VariantContextUtils.initializeMatchExps(filterNames, filterExpressions);
+        genotypeFilterExps = VariantContextUtils.initializeMatchExps(genotypeFilterNames, genotypeFilterExpressions);
+
+        VariantContextUtils.engine.get().setSilent(true);
+
+        initializeVcfWriter();
+    }
+
+    @Override
+    public void apply(final VariantContext variant, final ReadsContext readsContext, final ReferenceContext ref, final FeatureContext featureContext) {
+
+        final VariantContext vc1 = invalidatePreviousFilters ? (new VariantContextBuilder(variant)).filters(Collections.emptySet()).make() : variant;
```

Comment:
Use `VariantContextBuilder.unfiltered()` here instead of `filters(Collections.emptySet())`